// fastify-zod/build/__tests__/server.fixtures.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.swaggerOptions = exports.openApiOptions = exports.createTestServer = void 0;

var _fastify = _interopRequireDefault(require("fastify"));

var _httpErrors = require("http-errors");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const swaggerOptions = {
  routePrefix: `/swagger`,
  swagger: {
    info: {
      title: `Fastify Zod Test Server`,
      description: `Test Server for Fastify Zod`,
      version: `0.0.0`
    }
  },
  staticCSP: true,
  exposeRoute: true
};
exports.swaggerOptions = swaggerOptions;
const openApiOptions = {
  openapi: {
    info: {
      title: `Fastify Zod Test Server`,
      description: `Test Server for Fastify Zod`,
      version: `0.0.0`
    }
  },
  staticCSP: true,
  exposeRoute: true
};
exports.openApiOptions = openApiOptions;

const createTestServer = (fastifyOptions, registerOptions) => {
  const f = (0, _.register)((0, _fastify.default)(fastifyOptions), registerOptions);
  const state = {
    todoItems: []
  };
  f.zod.get(`/item`, {
    operationId: `getTodoItems`,
    reply: {
      description: `The list of Todo Items`,
      key: `TodoItems`
    }
  }, async () => state);
  f.zod.get(`/item/grouped-by-status`, {
    operationId: `getTodoItemsGroupedByStatus`,
    reply: `TodoItemsGroupedByStatus`
  }, async () => ({
    done: state.todoItems.filter(item => item.state === `done`),
    inProgress: state.todoItems.filter(item => item.state === `in progress`),
    todo: state.todoItems.filter(item => item.state === `todo`)
  }));
  f.zod.post(`/item`, {
    operationId: `postTodoItem`,
    body: `TodoItem`,
    reply: `TodoItems`
  }, async ({
    body: nextItem
  }) => {
    if (state.todoItems.some(prevItem => prevItem.id === nextItem.id)) {
      throw new _httpErrors.BadRequest(`item already exists`);
    }

    state.todoItems = [...state.todoItems, nextItem];
    return state;
  });
  f.zod.put(`/item/:id`, {
    operationId: `putTodoItem`,
    body: `TodoItem`,
    params: `TodoItemId`,
    reply: `TodoItem`
  }, async ({
    params: {
      id
    },
    body: nextItem
  }) => {
    if (!state.todoItems.some(prevItem => prevItem.id === id)) {
      throw new _httpErrors.NotFound(`no such item`);
    }

    state.todoItems = state.todoItems.map(prevItem => prevItem.id === id ? nextItem : prevItem);
    return nextItem;
  });
  f.zod.get(`/42`, {
    operationId: `getFortyTwo`,
    reply: `FortyTwo`
  }, async () => 42);
  return f;
};

exports.createTestServer = createTestServer;
